### PR TITLE
feat(sdk-core): add payInvoice to lightning object

### PR DESCRIPTION
**Pull request by sj3014**
- it pays a given lightning invoice

Ticket: BG-54082

### CODE REVIEW

**Comment by sj3014** [comment]
This pr will be merged into master once #2636 is merged

**Review by OttoAllmendinger — DISMISSED** [review]
please add @BitGo/btc-team as codeowners

**Review by tylerlevine — APPROVED** [review]
nice work

**Pull request merged**
